neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,103 @@ def _split(self, data_inputs):
         return splitted_data_inputs
 
 
+class WalkForwardTimeSeriesCrossValidation(BaseCrossValidation):
+    """
+    This will change perform a walk forward cross validation by performing a forward rolling split.
+    All the training split will start at the start of the time series but the end will increase toward
+    the end at each forward step. For the validation split it will start after each training split and
+    can have a delay to test prediction according to some delay.
+
+    :param scoring_function: scoring function use to validate performance if it is not None, by default r2_score,
+    :param window_size: the size of the time step taken between each forward roll, by default 3.
+    :param initial_window_number: the size of the first training split length in term of number of window a window is defined by the window_size, by default 3.
+    :param window_delay_size: the size of the delay between the end of the training split and the validation split, by default 0.
+    :param joiner the joiner callable that can join the different result together.
+    :return: WalkForwardTimeSeriesCrossValidation instance.
+    """
```

Comment:
Thanks so much for having a nice docstring for the documentation website. 

I'd at least suggest adding a bit more clarity, especially specifying the expected data_input shape to do the split correctly on the dimensions, people might not assume that the shape is of `[batch_size, total_time_steps, n_features]` or something like that. Dimensions' shapes of inputs and outputs are always the most important thing to document since the data being processed are simple arrays and that there are no obvious ways to label it's type yet (e.g.: with DTOs).

Reply:
Yeah, I agree with you.